ClangSharpPInvokeGenerator/Cursors/Decls/StructDecl.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics;
+using ClangSharp;
+
+namespace ClangSharpPInvokeGenerator
+{
+    internal sealed class StructDecl : Decl
+    {
+        private readonly List<FieldDecl> _fieldDecls;
+
+        public StructDecl(CXCursor handle, Cursor parent) : base(handle, parent)
+        {
+            Debug.Assert(handle.Kind == CXCursorKind.CXCursor_StructDecl);
+            _fieldDecls = new List<FieldDecl>();
+        }
+
+        public IReadOnlyList<FieldDecl> FieldDecls => _fieldDecls;
+
+        protected override CXChildVisitResult VisitChildren(CXCursor childHandle, CXCursor handle, CXClientData clientData)
+        {
+            ValidateVisit(ref handle);
+
+            switch (childHandle.Kind)
+            {
+                case CXCursorKind.CXCursor_StructDecl:
+                {
+                    return GetOrAddChild<StructDecl>(childHandle).Visit(clientData);
+                }
+
+                case CXCursorKind.CXCursor_UnionDecl:
+                {
+                    return GetOrAddChild<UnionDecl>(childHandle).Visit(clientData);
+                }
+
+                case CXCursorKind.CXCursor_EnumDecl:
+                {
+                    return GetOrAddChild<EnumDecl>(childHandle).Visit(clientData);
+                }
+
+                case CXCursorKind.CXCursor_FieldDecl:
+                {
+                    var fieldDecl = GetOrAddChild<FieldDecl>(childHandle);
```

Comment:
Most code paths in these types aren't consumed by generating LLVMSharp or ClangSharp, but they can be used for generating things like the Windows SDK.

Child cursors that represent useful information for a given cursor kind can be exposed directly to the writer, such as the `FieldDecl` here. The writer can then handle that data first before falling back to normal processing for any remaining children.